docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
+### Prepare to run
 
-    static final String MONEY_TRANSFER_TASK_QUEUE = "MONEY_TRANSFER_TASK_QUEUE";
-}
+Start running a [Temporal Service](https://docs.temporal.io/clusters) by launching it from a Terminal session, as discussed in the [Getting Started tutorial](https://learn.temporal.io/getting_started/java/dev_environment/):
+
+
+```bash
+temporal server start-dev
 ```
+
+If a Service is already running, you do not (and cannot) have to re-launch it again.
+If you are running a Unix-like system, you can check for this by issuing `pgrep`:
+
+
+```bash
+pgrep -fl temporal
+```
+
+### Workflow Execution
+
+When you _start_ a Workflow, you tell the Temporal Server, "Track the state of a Workflow that uses this method signature."
+This tracked state is called a Workflow Execution, and follows the code from the Workflow and Activity definitions you just saw through their lifetimes.
+The Workflow Execution represents a complete history of the Workflow's lifetime, including its intermediate states, its retries, and so forth.
+
+The Service breaks down the Workflow Execution into pieces, called Tasks.
+There are Workflow Tasks and Activity Tasks.
+Workers look for those Tasks on "Task Queues" (called "polling" for Tasks).
+That's why when you set up a Java Worker, you must tell it the Task Queue it should use to look for work.
+
+A Worker runs each piece of the Workflow separately, Task by Task.
+Whether that Task runs perfectly or it encounters issues, the Worker relays the execution events and results back to the Server.
+This is recorded into the Event History for the Workflow Execution.
+
+Workflow Executions "live" on the Temporal Server.
+In the most basic form, a Workflow Execution consists of a set of set-up data and an Event history.
+You initiate a Workflow Execution with the name of a Workflow Definition (the `MoneyTransferWorkflow` class), a Task Queue name to use, and optional input data.
+Each Workflow Execution has an identifier.
+If you do not supply one, the Service creates a random unique ID to use instead.
+
+### The Transfer application
+
+The "money transfer" app is a small Java program that creates a Workflow Execution.
+In a real deployments, you might invoke similar code when someone submits a form, presses a button, or visits a certain URL.
+Here is the code that performs this work:
+
+<!--SNIPSTART money-transfer-java-initiate-transfer {"selectedLines": ["31-70"]}-->
 <!--SNIPEND-->
 
-Run the `TransferMoneyWorker` class from IntelliJ, or run the following command from the project root in separate terminal:
+This code builds a Temporal Client, which is a type that can communicate with the Temporal Service.
+The Client is used to create a new Workflow stub that points to the Workflow method.
+With the details for the transfer collected together in a single instance (`transaction`), the `WorkflowClient` type creates a new Workflow Execution using the stub and the `transaction` instance.
+The execution runs asynchronously.
```

Comment:
I can't see the code but if it's using `activities.withdraw` and not a Promise (which I haven't heard you mention) it's not asynchronous.

Reply:
Updating the wording. Thanks.